sdk/storage/Azure.Storage.Files.DataLake/src/Models/DataLakeRetentionPolicy.cs

```diff
@@ -0,0 +1,26 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Azure.Storage.Files.DataLake.Models
+{
+    /// <summary>
+    /// The retention policy which determines how long the associated data should persist.
+    /// </summary>
+    public class DataLakeRetentionPolicy
+    {
+        /// <summary>
+        /// Indicates whether a retention policy is enabled for the storage service.
+        /// </summary>
+        public bool Enabled { get; set; }
+
+        /// <summary>
+        /// Indicates the number of days that metrics or logging or soft-deleted data should be retained. All data older than this value will be deleted.
```

Comment:
maybe we shouldn't mention what this setting can affect. I.e. don't mention logging soft-deleted etc. this can be derived from where property of that type is nested.
I made a mistake and read this comment first before realizing that property of this type exists in many places and made me wonder why I'd set soft delete retention in metrics configuration.
Somebody who's full text searching codebase/docs may fall in such trap.
Besides, if this class is used in more places we'd most likely forget to update this doc.

Reply:
This was copied almost exactly from `BlobRetentionPolicy`.